mesh/mesh_connections.hpp

```diff
@@ -0,0 +1,34 @@
+class MeshConnections
+{
+   private:
+   Mesh *mesh;
+   Array<Array<Table*>> T;
+
+   public:
+   struct EntityID
+   {
+      int dim;
+      int id;
+      bool boundary;
+   }
+
+   struct EntityIDs
+   {
+      int dim;
+      Array<int> ids;
+      bool boundary;
+   }
+
+   //Dimension independent interface
+   bool        IsChild(EntityID parent, EntityID child);
+   Array<int> &Neighbors(EntityID entity, int shared_dim);
```

Comment:
So, I unfortunately missed the discussion where this was initially discussed. Therefore, I'm trying to glean what this class is meant to provide based on the class names and comments from here. However, it seems like this class is meant to provide a nice way to query what elements / faces / edges / vertices neighbors are. So, it seems like this class could be very useful in building up graph analysis on mesh related data. Therefore, I have a few questions related to this class.

1. Does this class easily allow us to query how a boundary element / quantities are mapped to the normal mesh quantities. I currently have no idea how to do this in a sane manner using the existing mesh member functions, and I'd really like there to be one...
2. Would these methods return local element / face / edge / vertex numbering / neighbors? I'm only asking as I'd also be interested in the global info as well for neighbors that cross processor boundaries. Since, this info would make it simpler for me to build up a global graph to perform analysis such as connected component analysis on the global system. I guess if I just had the global IDs for the elements / vertices and had an easy to query which processors those were shared on then I could still do the global graph analysis that I might want to do.

I'm really interested in the answers to question 2 given part of my PhD thesis made us of these sorts of analysis to pry out non-intuitive insights into complicated problems.

Reply:
Yes, the intent is to be able to query on boundary entities as well.  I'll put some real documentation up to make this clear now that we converging on the interface.

This object is entirely local however as it interfaces with Mesh and not ParMesh.  I have some ideas for similar features for ParMesh, but that will have to wait for another day.

Reply:
@rcarson3 an incoming update with a starting point for the documentation should make this more clear.  Stay tuned today.